sensor/kubernetes/sensor/sensor.go

```diff
@@ -74,7 +74,13 @@ func CreateSensor(cfg *CreateOptions) (*sensor.Sensor, error) {
 	log.Infof("Install method: %q", helmManagedConfig.GetManagedBy())
 	installmethod.Set(helmManagedConfig.GetManagedBy())
 
-	deploymentIdentification := FetchDeploymentIdentification(context.Background(), cfg.k8sClient.Kubernetes())
+	k8sFetcher := cfg.k8sClient.Kubernetes()
```

Comment:
Nitpick: instead of introducing a variable here, maybe it would be simpler to set `cfg.realK8sClient` to `cfg.k8sClient` if the former is `nil` and then just use it below? 🤔

Reply:
done